src/SwiftRuntimeLibrary/src/SwiftCore.xml

```diff
@@ -0,0 +1,50 @@
+            <typedeclaration kind="struct" name="Swift.Int8" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Int16" entityType="Scalar" blittable="true" size="2" stride="2">
+            <typedeclaration kind="struct" name="Swift.Int16" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="UInt16" entityType="Scalar" blittable="true" size="2" stride="2">
+            <typedeclaration kind="struct" name="Swift.UInt16" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Int32" entityType="Scalar" blittable="true" size="4" stride="4">
+            <typedeclaration kind="struct" name="Swift.Int32" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="UInt32" entityType="Scalar" blittable="true" size="4" stride="4">
+            <typedeclaration kind="struct" name="Swift.UInt32" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Int64" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.Int64" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="UInt64" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.UInt64" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Single" entityType="Scalar" blittable="true" size="4" stride="4">
+            <typedeclaration kind="struct" name="Swift.Float" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="Double" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.Double" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="IntPtr" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.Int" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="UIntPtr" entityType="Scalar" blittable="true" size="8" stride="8">
+            <typedeclaration kind="struct" name="Swift.UInt" module="Swift" />
+        </entity>
+        <entity managedNameSpace="System" managedTypeName="bool" entityType="Scalar" blittable="false" size="4" stride="4">
+            <typedeclaration kind="struct" name="Swift.Bool" module="Swift" />
+        </entity>
+        <entity managedNameSpace="SwiftRuntimeLibrary" managedTypeName="UnsafeRawPointer" entityType="Scalar" blittable="false" size="0" stride="0">
+            <typedeclaration kind="struct" name="Swift.UnsafeRawPointer" module="Swift" />
+        </entity>
+        <entity managedNameSpace="SwiftRuntimeLibrary" managedTypeName="UnsafeMutableRawPointer" entityType="Scalar" blittable="false" size="0" stride="0">
```

Comment:
Can Swift have zero sized data types? If so, are they exposed in the stable ABI? I don't recall standard C or C++ permitting zero sized data types, but could be mistaken. Either way, are these really size 0?

Reply:
Look at the previous comment, I suggest excluding it until it becomes necessary.

Reply:
Just as an FYI, Swift supports 0 length value types.